### PR TITLE
TelemetryMessage should be able to take bytes and not serialize them

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/IotHubBaseClient.cs [hunk 1]
```diff
@@ -151,8 +151,8 @@ public async Task SendTelemetryAsync(TelemetryMessage message, CancellationToken
             }
 
             message.PayloadConvention = _clientOptions.PayloadConvention;
-            message.ContentType = _clientOptions.PayloadConvention.ContentType;
-            message.ContentEncoding = _clientOptions.PayloadConvention.ContentEncoding;
+            message.ContentType ??= _clientOptions.PayloadConvention.ContentType;
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:154]: Respect the value the user set, if set.

File: iothub/device/src/Messaging/TelemetryMessage.cs [hunk 2]
```diff
@@ -29,6 +29,15 @@ public TelemetryMessage(object payload)
             Payload = payload;
         }
 
+        /// <summary>
+        /// Creates an instance of this class with the specified binary payload.
+        /// </summary>
+        /// <param name="binaryPayload">The binary payload to send.</param>
+        public TelemetryMessage(byte[] binaryPayload)
```
Comment (drwill-ms) [inline iothub/device/src/Messaging/TelemetryMessage.cs:36]: Does not yield an ambiguity compiler error as expected, so we offer this overload for clarity.

@rido-min

File: iothub/device/src/Messaging/TelemetryMessage.cs [hunk 3]
```diff
@@ -119,7 +128,7 @@ public string MessageSchema
         public string ContentType
         {
             get => GetSystemProperty<string>(MessageSystemPropertyNames.ContentType);
-            protected internal set => SystemProperties[MessageSystemPropertyNames.ContentType] = value;
+            set => SystemProperties[MessageSystemPropertyNames.ContentType] = value;
```
Comment (drwill-ms) [inline iothub/device/src/Messaging/TelemetryMessage.cs:131]: Allow user to set it if they want to, especially for binary payloads.

File: iothub/device/src/Messaging/TelemetryMessage.cs [hunk 4]
```diff
@@ -236,9 +245,18 @@ public void SetAsSecurityMessage()
...
+        /// Gets the payload as a byte array, serialized and encoded if necessary.
+        /// </summary>
+        /// <remarks>
+        /// If needed, serialization uses Newtonsoft.Json and encoding is UTF8.
+        /// </remarks>
+        /// <returns>A payload as a byte array.</returns>
         internal byte[] GetPayloadObjectBytes()
         {
-            return PayloadConvention.GetObjectBytes(Payload);
+            return Payload is byte[] payloadAsByteArray
```
Comment (drwill-ms) [inline iothub/device/src/Messaging/TelemetryMessage.cs:257]: Don't serialize if byte[].

File: iothub/device/tests/TelemetryMessageTests.cs [hunk 5]
```diff
@@ -16,9 +16,22 @@ public class TelemetryMessageTests
         public void ConstructorTakingPayloadTest()
         {
             const string payloadString = "Hello, World!";
-            byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadString);
+            var msg = new TelemetryMessage(payloadString);
+            msg.Payload.Should().BeEquivalentTo(payloadString);
+
+            msg.GetPayloadObjectBytes().Should().BeEquivalentTo(DefaultPayloadConvention.Instance.GetObjectBytes(payloadString));
+        }
+        [TestMethod]
```
Comment (patilsnr) [inline iothub/device/tests/TelemetryMessageTests.cs:24]: nit: spacing

**Review by patilsnr — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]
lgtm

**Pull request merged**